File: clion-plugin/src/main/java/com/linkyourbin/alfred/UpdateActiveDevEnvProfileAction.java
package com.linkyourbin.alfred;

import com.intellij.openapi.actionSystem.AnAction;
import com.intellij.openapi.actionSystem.AnActionEvent;
import com.intellij.openapi.project.Project;
import com.intellij.openapi.ui.Messages;

import java.nio.file.Path;
import java.util.List;

public final class UpdateActiveDevEnvProfileAction extends AnAction {
    @Override
    public void actionPerformed(AnActionEvent event) {
        Project project = event.getProject();
        if (project == null || project.getBasePath() == null) {
            Messages.showErrorDialog("Open a project first.", DevEnvProfiles.TITLE);
            return;
        }

        try {
            Path root = DevEnvProfiles.root(project);
            DevEnvProfiles.migrateProjectProfiles(root);
            String active = DevEnvProfiles.activeProfileName();
            if (active.isBlank()) {
                List<Path> profileDirs = DevEnvProfiles.profileDirs(DevEnvProfiles.profiles());
                if (profileDirs.isEmpty()) {
                    Messages.showInfoMessage(project, "No profiles found in " + DevEnvProfiles.profiles(), DevEnvProfiles.TITLE);
                    return;
                }

                String[] names = profileDirs.stream().map(path -> path.getFileName().toString()).toArray(String[]::new);
                active = Messages.showEditableChooseDialog(
                        "Choose profile to update:",
                        DevEnvProfiles.TITLE,
                        Messages.getQuestionIcon(),
                        names,
                        names[0],
                        null
                );
                if (active == null || active.isBlank()) {
                    return;
                }
                DevEnvProfiles.saveProfile(root, active);
            } else {
                DevEnvProfiles.updateActiveProfile(root);
            }
            Messages.showInfoMessage(project, "Updated active profile: " + active, DevEnvProfiles.TITLE);
        } catch (Exception e) {
            Messages.showErrorDialog(project, e.getMessage(), DevEnvProfiles.TITLE);
        }
    }
}

File: clion-plugin/src/main/kotlin/com/linkyourbin/alfred/UpdateActiveDevEnvProfileAction.kt
package com.linkyourbin.alfred

import com.intellij.openapi.actionSystem.AnAction
import com.intellij.openapi.actionSystem.AnActionEvent
import com.intellij.openapi.ui.InputValidator
import com.intellij.openapi.ui.Messages

class UpdateActiveDevEnvProfileAction : AnAction() {
    override fun actionPerformed(event: AnActionEvent) {
        val project = event.project
        if (project?.basePath == null) {
            Messages.showErrorDialog("Open a project first.", DevEnvProfiles.TITLE)
            return
        }

        try {
            val root = DevEnvProfiles.root(project)
            DevEnvProfiles.migrateProjectProfiles(root)
            var active = DevEnvProfiles.activeProfileName()
            if (active.isBlank()) {
                val profileDirs = DevEnvProfiles.profileDirs(DevEnvProfiles.profiles())
                if (profileDirs.isEmpty()) {
                    Messages.showInfoMessage(project, "No profiles found in ${DevEnvProfiles.profiles()}", DevEnvProfiles.TITLE)
                    return
                }

                val names = profileDirs.map { it.fileName.toString() }.toTypedArray()
                val selected = Messages.showEditableChooseDialog(
                    "Choose profile to update:",
                    DevEnvProfiles.TITLE,
                    Messages.getQuestionIcon(),
                    names,
                    names[0],
                    null as InputValidator?,
                )
                if (selected.isNullOrBlank()) {
                    return
                }
                active = selected
                DevEnvProfiles.saveProfile(root, active)
            } else {
                DevEnvProfiles.updateActiveProfile(root)
            }
            Messages.showInfoMessage(project, "Updated active profile: $active", DevEnvProfiles.TITLE)
        } catch (e: Exception) {
            Messages.showErrorDialog(project, e.message ?: e.toString(), DevEnvProfiles.TITLE)
        }
    }
}
